common/metrics/opentelemetry_provider.go

```diff
@@ -31,37 +31,36 @@ import (
 
 	"go.opentelemetry.io/otel/exporters/prometheus"
 	"go.opentelemetry.io/otel/metric"
-	"go.opentelemetry.io/otel/sdk/export/metric/aggregation"
 	controller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
+	"go.opentelemetry.io/otel/sdk/metric/export/aggregation"
 	processor "go.opentelemetry.io/otel/sdk/metric/processor/basic"
 	"go.opentelemetry.io/otel/sdk/resource"
 
 	"go.temporal.io/server/common/log"
 	"go.temporal.io/server/common/log/tag"
 )
 
-var _ OpentelemetryMustProvider = (*opentelemetryMustProviderImpl)(nil)
+var _ OpentelemetryProvider = (*opentelemetryProviderImpl)(nil)
 
 type (
-	OpentelemetryMustProvider interface {
+	OpentelemetryProvider interface {
 		Stop(logger log.Logger)
-		GetMeterMust() metric.MeterMust
```

Comment:
`MeterMust` wrapper was removed in v0.28.0. and methods on metric.Meter no longer return errors. Errors are returned when creating the counter, histogram, gauge. See scope.go file below.